openmc/material.py

```diff
@@ -0,0 +1,1109 @@
+            if Z >= 89:
+                self.depletable = True
+
+        self._nuclides.append((nuclide, percent, percent_type))
+
+    def remove_nuclide(self, nuclide):
+        """Remove a nuclide from the material
+
+        Parameters
+        ----------
+        nuclide : str
+            Nuclide to remove
+
+        """
+        cv.check_type('nuclide', nuclide, str)
+
+        # If the Material contains the Nuclide, delete it
+        for nuc in self._nuclides:
+            if nuclide == nuc[0]:
+                self._nuclides.remove(nuc)
+                break
+
+    def add_macroscopic(self, macroscopic):
+        """Add a macroscopic to the material.  This will also set the
+        density of the material to 1.0, unless it has been otherwise set,
+        as a default for Macroscopic cross sections.
+
+        Parameters
+        ----------
+        macroscopic : str
+            Macroscopic to add
+
+        """
+
+        # Ensure no nuclides, elements, or sab are added since these would be
+        # incompatible with macroscopics
+        if self._nuclides or self._sab:
+            msg = 'Unable to add a Macroscopic data set to Material ID="{}" ' \
+                  'with a macroscopic value "{}" as an incompatible data ' \
+                  'member (i.e., nuclide or S(a,b) table) ' \
+                  'has already been added'.format(self._id, macroscopic)
+            raise ValueError(msg)
+
+        if not isinstance(macroscopic, str):
+            msg = 'Unable to add a Macroscopic to Material ID="{}" with a ' \
+                  'non-string value "{}"'.format(self._id, macroscopic)
+            raise ValueError(msg)
+
+        if self._macroscopic is None:
+            self._macroscopic = macroscopic
+        else:
+            msg = 'Unable to add a Macroscopic to Material ID="{}". ' \
+                  'Only one Macroscopic allowed per ' \
+                  'Material.'.format(self._id)
+            raise ValueError(msg)
+
+        # Generally speaking, the density for a macroscopic object will
+        # be 1.0. Therefore, lets set density to 1.0 so that the user
+        # doesnt need to set it unless its needed.
+        # Of course, if the user has already set a value of density,
+        # then we will not override it.
+        if self._density is None:
+            self.set_density('macro', 1.0)
+
+    def remove_macroscopic(self, macroscopic):
+        """Remove a macroscopic from the material
+
+        Parameters
+        ----------
+        macroscopic : str
+            Macroscopic to remove
+
+        """
+
+        if not isinstance(macroscopic, str):
+            msg = 'Unable to remove a Macroscopic "{}" in Material ID="{}" ' \
+                  'since it is not a string'.format(self._id, macroscopic)
+            raise ValueError(msg)
+
+        # If the Material contains the Macroscopic, delete it
+        if macroscopic == self._macroscopic:
+            self._macroscopic = None
```

Comment:
under what circumstances is it not an error to try to call remove_macroscopic when self._macroscopic isn't the parameter? Without knowing what the intended usage looks like, I was expecting either `_macroscopic` to be a dictionary/set of strings or for this method to raise if this condition is false - if nothing else a comment would help